stage0/src/sev.rs

```diff
@@ -124,59 +331,39 @@ pub fn validate_memory(e820_table: &[BootE820Entry], encrypted: u64) {
             continue;
         }
 
-        let mut range = PhysFrame::<Size4KiB>::range(
-            PhysFrame::from_start_address(PhysAddr::new(align_up(
-                entry.addr() as u64,
-                Size4KiB::SIZE,
-            )))
-            .unwrap(),
-            PhysFrame::from_start_address(PhysAddr::new(align_down(
-                (entry.addr() + entry.size()) as u64,
-                Size4KiB::SIZE,
-            )))
-            .unwrap(),
-        );
-
-        // Our temporary page table will only fit 512 4K frames at a time, so iterate in chunks of
-        // 512 (with the last one being smaller if needed)
-        while page_table
-            .iter_mut()
-            .map(|pte| {
-                if let Some(frame) = range.next() {
-                    pte.set_addr(frame.start_address() + encrypted, PageTableFlags::PRESENT);
-                    1
-                } else {
-                    0
-                }
-            })
-            .sum::<u64>()
-            > 0
+        // If the memory boundaries align with 2 MiB, start with that.
+        if entry.addr() as u64 % Size2MiB::SIZE == 0
+            && (entry.addr() + entry.size()) as u64 % Size2MiB::SIZE == 0
         {
-            tlb::flush_all();
-
-            if let Some(err) = page_table
-                .iter()
-                .enumerate()
-                .filter(|(_, entry)| !entry.is_unused())
-                .map(|(n, _)| {
-                    pvalidate(
-                        Size2MiB::SIZE as usize + (n * Size4KiB::SIZE as usize),
-                        SevPageSize::Page4KiB,
-                        Validation::Validated,
-                    )
-                })
-                .map(|result| match result {
-                    Err(InstructionError::ValidationStatusNotUpdated) => Ok(()),
-                    other => other,
-                })
-                .find(|result| result.is_err())
-            {
-                panic!("unexpected error: {:?}", err);
-            }
-
-            page_table.zero();
+            let range = PhysFrame::<Size2MiB>::range(
```

Comment:
I see a mix of `unwrap` and `expect` in the code below. Should all be `expect`?

Reply:
These unwraps should never fail as we've made sure the addresses are properly aligned. I'll add a comment.